Microsoft.TeamServices.Samples.Client/Release/ReleasesSample.cs

```diff
@@ -0,0 +1,512 @@
+            int releaseDefinitionId = releaseDefinitions.FirstOrDefault().Id;
+            Microsoft.VisualStudio.Services.ReleaseManagement.WebApi.Release release = CreateRelease(releaseClient, releaseDefinitionId, projectName);
+
+            ReleaseEnvironmentUpdateMetadata releaseEnvironmentUpdateMetadata = new ReleaseEnvironmentUpdateMetadata()
+            {
+                Status = EnvironmentStatus.InProgress
+            };
+
+            int releaseEnvironmentId = release.Environments.FirstOrDefault().Id;
+
+            // Abandon a release
+            ReleaseEnvironment releaseEnvironment = releaseClient.UpdateReleaseEnvironmentAsync(releaseEnvironmentUpdateMetadata, projectName, release.Id, releaseEnvironmentId).Result;
+            Console.WriteLine("{0} {1}", releaseEnvironment.Id.ToString().PadLeft(6), releaseEnvironment.Name);
+
+            return release;
+        }
+
+        [ClientSampleMethod]
+
+        public Microsoft.VisualStudio.Services.ReleaseManagement.WebApi.Release AbandonAnActiveRelease()
+        {
+            string projectName = ClientSampleHelpers.FindAnyProject(this.Context).Name;
+
+            // Get a release client instance
+            VssConnection connection = Context.Connection;
+            ReleaseHttpClient releaseClient = connection.GetClient<ReleaseHttpClient>();
+
+            List<ReleaseDefinition> releaseDefinitions = releaseClient.GetReleaseDefinitionsAsync(project: projectName).Result;
+            int releaseDefinitionId = releaseDefinitions.FirstOrDefault().Id;
+            Microsoft.VisualStudio.Services.ReleaseManagement.WebApi.Release release = CreateRelease(releaseClient, releaseDefinitionId, projectName);
+
+            ReleaseUpdateMetadata releaseUpdateMetadata = new ReleaseUpdateMetadata()
+            {
+                Comment = "Abandon the release",
+                Status = ReleaseStatus.Abandoned
+            };
+
+            // Abandon a release
+            Microsoft.VisualStudio.Services.ReleaseManagement.WebApi.Release updatedRelease = releaseClient.UpdateReleaseResourceAsync(releaseUpdateMetadata, projectName, release.Id).Result;
+            Console.WriteLine("{0} {1}", updatedRelease.Id.ToString().PadLeft(6), updatedRelease.Name);
+
+            return release;
+        }
+
+        [ClientSampleMethod]
+
+        public IEnumerable<ReleaseApproval> ListAllPendingApprovals()
+        {
+            string projectName = ClientSampleHelpers.FindAnyProject(this.Context).Name;
+
+            // Get a release client instance
+            VssConnection connection = Context.Connection;
+            ReleaseHttpClient2 releaseClient = connection.GetClient<ReleaseHttpClient2>();
+
+            List<ReleaseApproval> releaseApprovals = new List<ReleaseApproval>();
+
+            // Iterate (as needed) to get the full set of approvals
+            int continuationToken = 0;
```

Comment:
This block is to handle pagination. Similar pattern used else where needed pagination.